peer/brontide.go

```diff
@@ -1663,8 +1664,25 @@ func (p *Brontide) handleCustomMessage(msg *lnwire.Custom) error {
 // isActiveChannel returns true if the provided channel id is active, otherwise
 // returns false.
 func (p *Brontide) isActiveChannel(chanID lnwire.ChannelID) bool {
-	_, ok := p.activeChannels.Load(chanID)
-	return ok
+	// The channel would be nil if,
+	// - the channel doesn't exist, or,
+	// - the channel exists, but is pending. In this case, we don't
+	//   consider this channel active.
+	channel, _ := p.activeChannels.Load(chanID)
+
+	return channel != nil
```

Comment:
We do we care about nil-ness here? We should just check the second arg to see if it exists or not. 

I know we use the `ok && nil` trick in a few other places, but IMO we should shy away from that in favor of _not_ storing `nil` values in a map to mean a certain state. 

Alternatively, we can make a very simple wrapper struct around the channel itself to be able to thread through another state to fill this new perceived gap.

Reply:
Agree we should avoid the implicit usage of `nil` in map as well. Added a commit before this one to remove it.